lib/matplotlib/axes/_base.py

```diff
@@ -29,6 +30,61 @@
 _log = logging.getLogger(__name__)
 
 
+class _axis_method_wrapper:
+    """
+    Helper to generate Axes methods wrapping Axis methods.
+
+    After ::
+
+        get_foo = _axis_method_wrapper("xaxis", "get_bar")
+
+    (in the body of a class) ``get_foo`` is a method that forwards it arguments
+    to the ``get_bar`` method of the ``xaxis`` attribute, and gets its
+    signature and docstring from ``Axis.get_bar``.
+
+    The docstring of ``get_foo`` is built by replacing "this Axis" by "the
```

Comment:
I'm slightly worried that subsequent edits on the docstring will break this clever replacements. People usually won't know that "this Axis" has a special meaning. They could easily write:
- `this axis`
- ``this `.Axis` ``
- `the axis`

I'm afraid that we can't do much in terms of documenting this for docstring authors or warning upon processing. But let's at least include the first two variants for replacement.

Reply:
I'd rather keep things consistent.  Note that there is an assertion check that the correct substring is present, with a hopefully clear error message -- and the check occurs at import time, so we'd easily catch it in CI.

Reply:
Ok, I did not realize the assert.